### PR TITLE
[music] MUSIC_UTILS::GetItemsForPlayList: Improve playlist sort 

### PR DESCRIPTION
**Pull request by ksooo**
... algorithm to use ByTrackNumber / ByArtistThenYear sort method if available for the folder to add to the playlist.

This decouples the sort method used for playlist from what the user has set as a sort method in the respective windows.

Playlists containing a single albums should always be sorted by track number now.
Playlist containing multiple albums are sorted by artist, then year.

If the folder to play does neither support by track number, nor by artist then year, the sort method set for the respective window will be used.

I tested the changes on macOS and Android, latest master.

@the-black-eagle could you do a runtime test to ensure behavior is as expected?
@enen92 whenever you find some time for a code review. The logic is now the same as implemented in the continue watching video folders PR.

### CODE REVIEW

**Comment by ksooo** [comment]
Oh, just noticed that this PR has a compile time dependency on #22107 - so no runtime-test possible yet. But code review makes sense anyway.

File: xbmc/music/MusicUtils.cpp [hunk 1]
```diff
@@ -444,6 +444,55 @@ class CAsyncGetItemsForPlaylist : public IRunnable
...
+      int lastAlbumId = -1;
+      for (const auto& item : items)
+      {
+        if (item->HasMusicInfoTag())
+        {
+          const auto tag = item->GetMusicInfoTag();
+          if (lastAlbumId != -1 && tag->GetAlbumId() != lastAlbumId)
+          {
+            // First choice for folders containing multiple albums
+            SortDescription desc = sortDescription;
```
Comment (enen92) [inline xbmc/music/MusicUtils.cpp:480]: Wouldn't it make more sense to make `sortDescTrackNumber` a bit more generic and just reuse it here as well just like you did above?
Or to simply remove the const modifier in the iterator and set the properties on the sortDescription itself?

Comment (ksooo) [reply xbmc/music/MusicUtils.cpp:480]: > Or to simply remove the const modifier in the iterator and set the properties on the sortDescription itself?

 Yes, will change that.

Comment (ksooo) [reply xbmc/music/MusicUtils.cpp]: done

**Comment by ksooo** [comment]
Rebased to latest master, should compile now and can be runtime-tested, @the-black-eagle if you find some time.

**Comment by ksooo** [comment]
@enen92 could you please re-view and approve in case you are fine with the code changes now.

**Review by enen92 — APPROVED** [review]
I don't see anything wrong. Risk should not be high for this one.

**Comment by ksooo** [comment]
@the-black-eagle if you find some time, this is ready for a runtime-test now.

File: xbmc/music/MusicUtils.cpp [hunk 2]
```diff
@@ -444,6 +444,54 @@ class CAsyncGetItemsForPlaylist : public IRunnable
...
+      for (const auto& item : items)
+      {
+        if (item->HasMusicInfoTag())
+        {
+          const auto tag = item->GetMusicInfoTag();
+          if (lastAlbumId != -1 && tag->GetAlbumId() != lastAlbumId)
+          {
+            // First choice for folders containing multiple albums
+            sortDescription.sortOrder = SortOrderAscending;
+            return sortDescription;
```
Comment (garbear) [inline xbmc/music/MusicUtils.cpp:481]: It's generally nice to avoid early returns, though here the logic is easy enough to follow.

**Review by garbear — APPROVED** [review]
I see no issues in the code, and agree with @enen92 that it's low risk.

**Pull request merged**